addons/stock/models/stock_location.py

```diff
@@ -167,6 +167,12 @@ def _onchange_usage(self):
         if self.usage not in ('internal', 'inventory'):
             self.scrap_location = False
 
+    @api.constrains('scrap_location')
+    def _check_scrap_location(self):
+        for record in self:
+            if record.scrap_location and self.env['stock.picking.type'].search([('code', '=', 'mrp_operation'), ('default_location_dest_id', '=', record.id)]):
+                raise ValidationError(_("You cannot set a location of type Scrap, to a manufacturing type operation."))
```

Comment:
Sorry, the English is this will also need to be updated to make sense. `You cannot set a location as a scrap location when it assigned as a destination location for a manufacturing type operation.`